### PR TITLE
Migrate the stack with terraform 0.12

### PR DESCRIPTION
**Pull request by sdurrheimer**

### CODE REVIEW

File: pipeline/prometheus.yml [hunk 1]
```diff
@@ -49,19 +49,22 @@ resource_types:
...
 
 - name: tfstate
   type: terraform
   source:
-    storage:
+    env_name: ((env))
+    backend_type: s3
+    backend_config:
       bucket: ((terraform_storage_bucket_name))
-      bucket_path: ((terraform_storage_bucket_path))
```
Comment (talset) [inline pipeline/prometheus.yml]: terraform_storage_bucket_path param should be removed from pipeline variable and the readme too

File: terraform/module-prometheus/outputs.tf [hunk 2]
```diff
@@ -13,35 +13,36 @@
 #}
 
 output "prometheus_eip" {
-  value = "${element(aws_eip.prometheus.*.public_ip, 0)}"
+  value = element(aws_eip.prometheus.*.public_ip, 0)
```
Comment (talset) [inline terraform/module-prometheus/outputs.tf]: with terrafrom 0.12 we can now remove element(0) and use `aws_eip.prometheus[0].public_ip`

File: terraform/module-prometheus/outputs.tf [hunk 3]
```diff
@@ -13,35 +13,36 @@
...
 output "prometheus_instance_id" {
-  value = "${aws_instance.prometheus.id}"
+  value = aws_instance.prometheus[0].id
 }
 
 # workaround to handle count = 0 on output
 # https://github.com/hashicorp/terraform/issues/16726
 output "rds_address" {
-  value = "${element(concat(aws_db_instance.grafana.*.address, list("")), 0)}"
+  value = element(concat(aws_db_instance.grafana.*.address, [""]), 0)
```
Comment (talset) [inline terraform/module-prometheus/outputs.tf]: I think it is not needed anymore. You can remove the # workaround comment and use 

```
output "rds_address" {
  value = aws_db_instance.grafana[0].address
}
```

File: terraform/module-prometheus/prometheus.tf [hunk 4]
```diff
@@ -71,45 +71,42 @@ resource "aws_security_group_rule" "prometheus_to_himself_instance_metrics" {
...
 
-  vpc_security_group_ids = ["${compact(list(
-    "${var.bastion_sg_allow}",
-    "${aws_security_group.prometheus.id}",
-  ))}"]
+  vpc_security_group_ids = compact([var.bastion_sg_allow, aws_security_group.prometheus.id])
 
   # subnet_id = "${element(var.private_subnets_ids, count.index)}"
-  subnet_id = "${element(var.public_subnets_ids, count.index)}"
+  subnet_id = element(var.public_subnets_ids, count.index)
```
Comment (talset) [inline terraform/module-prometheus/prometheus.tf]: I stlll dont know in this case if the new format is better or not

`subnet_id = var.public_subnets_ids[count.index]`

File: terraform/module-prometheus/prometheus.tf [hunk 5]
```diff
@@ -71,45 +71,42 @@ resource "aws_security_group_rule" "prometheus_to_himself_instance_metrics" {
 ###
 
 resource "aws_instance" "prometheus" {
-  ami = "${data.aws_ami.debian.id}"
+  ami = data.aws_ami.debian.id
 
   # associate_public_ip_address = false
   count                = 1
```
Comment (talset) [inline terraform/module-prometheus/prometheus.tf]: I dont remember, why do we have `count = 1` without variable ?
Should we remove the count ? Should we add a variable ?

File: terraform/module-prometheus/prometheus.tf [hunk 6]
```diff
@@ -132,13 +129,14 @@ resource "aws_eip" "prometheus" {
 ###
 
 resource "aws_cloudwatch_metric_alarm" "recover-prometheus" {
+  count               = 1
```
Comment (talset) [inline terraform/module-prometheus/prometheus.tf]: liked to my comment on aws_instance `count = 1`

File: terraform/module-prometheus/prometheus.tf (hunk 5 above)
Comment (sdurrheimer) [reply terraform/module-prometheus/prometheus.tf]: `var.count` was used in name, tags etc, so declaring `count = 1` is mandatory in 0.12.

File: pipeline/prometheus.yml (hunk 1 above)
Comment (sdurrheimer) [reply pipeline/prometheus.yml]: Thx, missed that.

File: terraform/module-prometheus/outputs.tf (hunk 3 above)
Comment (sdurrheimer) [reply terraform/module-prometheus/outputs.tf]: Will this work if `count = 0` ?

Comment (talset) [reply terraform/module-prometheus/outputs.tf]: Yes I tested it on lemp

**Pull request merged**